src/StargateEventListener/Handlers/StargateEventHandlers.ts

```diff
@@ -0,0 +1,269 @@
+import { abi } from "thor-devkit"
+import { ThorClient } from "@vechain/sdk-network"
+import { Beat, Network } from "~Model"
+import { BloomUtils, debug, error } from "~Utils"
+import { ERROR_EVENTS } from "~Constants"
+import { getStargateNetworkConfig } from "~Constants/Constants/Staking"
+
+// NodeDelegated event ABI - event NodeDelegated(uint256 indexed nodeId, address indexed delegatee, bool indexed isDelegated)
+const NODE_DELEGATED_EVENT_ABI = {
+    anonymous: false,
+    inputs: [
+        { indexed: true, internalType: "uint256", name: "nodeId", type: "uint256" },
+        { indexed: true, internalType: "address", name: "delegatee", type: "address" },
+        { indexed: true, internalType: "bool", name: "isDelegated", type: "bool" },
+    ],
+    name: "NodeDelegated",
+    type: "event" as const,
+}
```

Comment:
This is probably not needed aswell, I think you can grab it from `~Generated/abi`